pkg/genericapiserver/openapi/openapi.go

```diff
@@ -0,0 +1,470 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/emicklei/go-restful"
+	"github.com/emicklei/go-restful/swagger"
+	"github.com/go-openapi/spec"
+)
+
+type Config struct {
+	SwaggerConfig *swagger.Config
+	// List of webservice's path prefixes to ignore
+	IgnorePrefixes []string
+}
+
+type openApi struct {
```

Comment:
style: API is an initialism so it should be capitalized - `openAPI`, `RegisterOpenAPIService`